cpp/src/join/key_remapping.cu

```diff
@@ -0,0 +1,651 @@
+      cudf::nullate::DYNAMIC{has_nulls}, preprocessed_build, preprocessed_build, compare_nulls};
+
+    using comparator_type = build_comparator<decltype(d_equal)>;
+    return std::make_unique<key_remap_table<comparator_type>>(build,
+                                                              preprocessed_build,
+                                                              comparator_type{d_equal},
+                                                              d_hasher,
+                                                              compare_nulls,
+                                                              load_factor,
+                                                              stream);
+  } else if (has_nested) {
+    auto const row_hasher = cudf::detail::row::hash::row_hasher{preprocessed_build};
+    auto const d_hasher   = row_hasher.device_hasher(cudf::nullate::DYNAMIC{has_nulls});
+    auto const d_equal =
+      self_equal.equal_to<true>(cudf::nullate::DYNAMIC{has_nulls}, compare_nulls);
+
+    using comparator_type = build_comparator<decltype(d_equal)>;
+    return std::make_unique<key_remap_table<comparator_type>>(build,
+                                                              preprocessed_build,
+                                                              comparator_type{d_equal},
+                                                              d_hasher,
+                                                              compare_nulls,
+                                                              load_factor,
+                                                              stream);
+  } else {
+    auto const row_hasher = cudf::detail::row::hash::row_hasher{preprocessed_build};
+    auto const d_hasher   = row_hasher.device_hasher(cudf::nullate::DYNAMIC{has_nulls});
+    auto const d_equal =
+      self_equal.equal_to<false>(cudf::nullate::DYNAMIC{has_nulls}, compare_nulls);
+
+    using comparator_type = build_comparator<decltype(d_equal)>;
+    return std::make_unique<key_remap_table<comparator_type>>(build,
+                                                              preprocessed_build,
+                                                              comparator_type{d_equal},
+                                                              d_hasher,
+                                                              compare_nulls,
+                                                              load_factor,
+                                                              stream);
+  }
+}
+
+}  // namespace
+
+/**
+ * @brief Implementation class for key_remapping
+ */
+class key_remapping_impl {
+ public:
+  key_remapping_impl(cudf::table_view const& build,
+                     cudf::null_equality compare_nulls,
+                     rmm::cuda_stream_view stream)
+    : _compare_nulls{compare_nulls},
+      _table{create_key_remap_table(build, compare_nulls, 0.5, stream)}
```

Comment:
Again, remove this hard coded load factor and use a global `constexpr` constant only when we call cuco API.